lib/cli/src/Cardano/CLI.hs

```diff
@@ -143,6 +146,15 @@ parseAllArgsWith cli args option = do
     getAllArgsOrExit :: Arguments -> Option -> IO (NE.NonEmpty String)
     getAllArgsOrExit = getAllArgsOrExitWith cli
 
+parseOptionalArg :: FromText a => Arguments -> Option -> IO (Maybe a)
+parseOptionalArg args option =
```

Comment:
@jonathanknowles Maybe this would have been useful to use here instead:

https://github.com/input-output-hk/cardano-wallet/blob/master/exe/wallet/Main.hs#L369-L377 ?

An example:

https://github.com/input-output-hk/cardano-wallet/blob/master/exe/wallet/Main.hs#L224

Reply:
> @jonathanknowles Maybe this would have been useful to use here instead:
> 
> https://github.com/input-output-hk/cardano-wallet/blob/master/exe/wallet/Main.hs#L369-L377 ?
> 
> An example:
> 
> https://github.com/input-output-hk/cardano-wallet/blob/master/exe/wallet/Main.hs#L224

@KtorZ I'll take a look at this later on to see whether we can use `optional` to simplify this code.